pennylane/ops/op_math/exp.py

```diff
@@ -43,6 +44,45 @@
 from .symbolicop import ScalarSymbolicOp
 
 
+@lru_cache
+def _get_has_generator_types(num_wires):
+    # Store operator classes with generators
+    has_generator_types = []
+    any_wires_types = []
+    for op_name in qml.ops.qubit.__all__:
+        op_class = getattr(qml.ops.qubit, op_name)
+        if op_class not in {qml.PauliRot, qml.PCPhase} and op_class.has_generator:
+            if op_class.num_wires == num_wires:
+                has_generator_types.append(op_class)
+            elif op_class.num_wires is None:
+                any_wires_types.append(op_class)
+
+    # prioritize types with that exact number of wires over types with any number of wires
+    # ie choose RZ before MultiRZ
+    return has_generator_types + any_wires_types
+
+
+def _find_equal_generator(base, coeff):
```

Comment:
Do we want to put a cache on this as well? I can imagine code like:
```python
if op.has_decomposition:
    op.decomposition()
```
both methods calls this helper function back to back.